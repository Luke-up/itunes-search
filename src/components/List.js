import { useEffect } from "react";
import { Container, Table, Button } from "react-bootstrap";
import React from "react";

//function renders a list of the search results
//music and books are called with this layout
function List(props) {
  const [searchResults, setSearchResults] = React.useState();

  //useEffect makes a get request to the server
  //function runs whenever the search values are updated
  useEffect(() => {
    function runFetch() {
      fetch(
        "/media?term=" +
          props.searchTerm +
          "&media=" +
          props.mediaType +
          "&attribute=" +
          props.attribute +
          "&limit=" +
          props.limit
      )
        .then((response) => response.json())
        .then((data) => {
          setSearchResults(data.results);
          console.log(data.results);
        });
    }
    runFetch();
  }, [props.mediaType, props.searchTerm, props.attribute, props.limit]);

  //function saves the object to the favourites array
  function addToFavourites(media) {
    if (!props.favourites) {
      props.setFavourites(media);
    } else {
      let newArray = props.favourites;
      newArray.push(media);
      props.setFavourites(newArray);
    }
  }

  //render maps the results of the search in a table format
  return (
    <Container className="bg-info rounded">
      <h1 className="text-start text-light">Hello {props.mediaType}</h1>
      {searchResults ? (
        <Table striped hover className="border text-start">
          <thead>
            <tr>
              <th>Title</th>
              <th>Artist</th>
              {props.mediaType === "music" ? <th>Album</th> : ""}
              <th>Genre</th>
              <th></th>
            </tr>
          </thead>
          <tbody>
            {searchResults.map((result) => {
              let key = result.collectionId;
              if (result.trackId) {
                key = result.trackId;
              }
              return (
                <tr key={key}>
                  {props.mediaType === "audiobook" ? (
                    <th>{result.collectionName}</th>
                  ) : (
                    <th>{result.trackName}</th>
                  )}

                  <th>{result.artistName}</th>
                  {props.mediaType === "music" ? (
                    <th>{result.collectionName}</th>
                  ) : (
                    ""
                  )}

                  <th>
                    {result.primaryGenreName
                      ? result.primaryGenreName
                      : result.genres[0]}
                  </th>
                  <th>
                    <Button
                      onClick={() => {
                        addToFavourites(result);
                      }}
                      className="py-1"
                    >
                      <i className="bi bi-hand-thumbs-up-fill"></i>
                    </Button>
                  </th>
                </tr>
              );
            })}
          </tbody>
        </Table>
      ) : (
        <p>We're looking, but haven't found anything yet :/</p>
      )}
    </Container>
  );
}

export default List;
