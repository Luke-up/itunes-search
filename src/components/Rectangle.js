import { useEffect } from "react";
import { Container, Row, Col, Image, Card, Button } from "react-bootstrap";
import React from "react";

//function renders layouts for movie type media
//rectangular cards in a grid format
function Rectangle(props) {
  const [searchResults, setSearchResults] = React.useState();

  //grid should increase in size when rendered on the 'all' page
  let xlSize = 4;
  let mdSize = 6;
  if (props.limit === 3) {
    xlSize = 12;
    mdSize = 12;
  }

  //useEffect contains function to make get request
  //useEffect runs whenever search button is clicked
  useEffect(() => {
    function runFetch() {
      fetch(
        "/media?term=" +
          props.searchTerm +
          "&media=" +
          props.mediaType +
          "&attribute=" +
          props.attribute +
          "&limit=" +
          props.limit
      )
        .then((response) => response.json())
        .then((data) => {
          setSearchResults(data.results);
        });
    }
    runFetch();
  }, [props.clicker]);

  //function pushes an object to the favourites array
  function addToFavourites(media) {
    if (!props.favourites) {
      props.setFavourites(media);
    } else {
      let newArray = props.favourites;
      newArray.push(media);
      props.setFavourites(newArray);
    }
  }

  //render of rectangular cards in a grid
  //function maps the results from the fetch request
  return (
    <Container className="bg-secondary rounded">
      <h1 className="text-start text-light">Hello {props.mediaType}</h1>
      {searchResults ? (
        <Row>
          {searchResults.map((result) => {
            return (
              <Col key={result.trackId} xl={xlSize} md={mdSize}>
                <Card className="my-2" id={result.trackId}>
                  <Row>
                    <Col xs={3}>
                      <Image
                        className="m-2"
                        alt={"Poster for: " + result.trackName}
                        src={result.artworkUrl100}
                      />
                    </Col>
                    <Col xs={9}>
                      <div className="mx-1 mt-2 row text-wrap">
                        <Col xs={10}>
                          <p>
                            <b>{result.trackName}</b>
                            <br />
                            {result.primaryGenreName}
                            <br />
                            {result.artistName}
                          </p>
                        </Col>
                        <Col xs={2}>
                          <div className="float-end">
                            {
                              <Button
                                onClick={() => {
                                  addToFavourites(result);
                                }}
                                className="py-1"
                              >
                                <i className="bi bi-hand-thumbs-up-fill"></i>
                              </Button>
                            }
                          </div>
                        </Col>
                      </div>
                    </Col>
                  </Row>
                </Card>
              </Col>
            );
          })}
        </Row>
      ) : (
        <p>We're looking, but haven't found anything yet :/</p>
      )}
    </Container>
  );
}
export default Rectangle;
