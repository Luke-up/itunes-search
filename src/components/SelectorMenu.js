import React, { useEffect } from "react";
import { Form } from "react-bootstrap";
import Attributes from "./Attributes.json";

//function renders a selection bar for each media type
//options are displayed for specific search criteria
function SelectorMenu(props) {
  const [optionsArray, setOptionsArray] = React.useState([{}]);

  //useeffect sets the array based on the type of media
  //the options are updated on media change
  useEffect(() => {
    if (props.mediaType === "movie") {
      setOptionsArray(Attributes.movie);
    } else if (props.mediaType === "podcast") {
      setOptionsArray(Attributes.podcast);
    } else if (props.mediaType === "music") {
      setOptionsArray(Attributes.music);
    } else if (props.mediaType === "audiobook") {
      setOptionsArray(Attributes.audiobook);
    } else if (props.mediaType === "shortFilm") {
      setOptionsArray(Attributes.shortFilm);
    } else if (props.mediaType === "tvShow") {
      setOptionsArray(Attributes.tvShow);
    } else {
      setOptionsArray("");
    }
  }, [optionsArray, props.mediaType]);

  //renders select dropdown in navigation bar
  return (
    <Form.Select onChange={(e) => props.setAttribute(e.target.value)}>
      {optionsArray ? (
        <option>Choose a search criteria</option>
      ) : (
        <option>Search by name</option>
      )}
      {optionsArray
        ? optionsArray.map((attribute) => {
            let key = 1;
            if (attribute.searchTerm) {
              key = attribute.searchTerm;
            }
            return (
              <option key={key} value={attribute.searchTerm}>
                {attribute.commonTerm}
              </option>
            );
          })
        : ""}
    </Form.Select>
  );
}

export default SelectorMenu;
