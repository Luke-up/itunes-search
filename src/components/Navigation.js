import { Container, Nav, Col, Row, Form } from "react-bootstrap";
import SelectorMenu from "./SelectorMenu.js";

//function returns a navigation bar at the top of the page
//input for search updates value on change
//attribute selection menu is a separate component that renders differently for each media type
function Navigation(props) {
  return (
    <Container className="pt-4 border rounded mx-auto my-3">
      <Row className="mb-4">
        <Col>
          <Form.Control onChange={(e) => props.setSearchTerm(e.target.value)} />
        </Col>
        <Col>
          <SelectorMenu
            mediaType={props.mediaType}
            setAttribute={props.setAttribute}
          />
        </Col>
      </Row>
      <Nav fill variant="tabs" defaultActiveKey="/all">
        <Nav.Item>
          <Nav.Link
            onClick={() => {
              props.setMediaType("all");
              props.setAttribute("other");
            }}
            eventKey="all"
          >
            All
          </Nav.Link>
        </Nav.Item>
        <Nav.Item>
          <Nav.Link
            onClick={() => {
              props.setMediaType("movie");
              props.setAttribute("other");
            }}
            eventKey="Movies"
          >
            Movies
          </Nav.Link>
        </Nav.Item>
        <Nav.Item>
          <Nav.Link
            onClick={() => {
              props.setMediaType("podcast");
              props.setAttribute("other");
            }}
            eventKey="Podcasts"
          >
            Podcasts
          </Nav.Link>
        </Nav.Item>
        <Nav.Item>
          <Nav.Link
            onClick={() => {
              props.setMediaType("music");
              props.setAttribute("other");
            }}
            eventKey="Music"
          >
            Music
          </Nav.Link>
        </Nav.Item>
        <Nav.Item>
          <Nav.Link
            onClick={() => {
              props.setMediaType("audiobook");
              props.setAttribute("other");
            }}
            eventKey="Audiobooks"
          >
            Audiobooks
          </Nav.Link>
        </Nav.Item>
        <Nav.Item>
          <Nav.Link
            onClick={() => {
              props.setMediaType("shortFilm");
              props.setAttribute("other");
            }}
            eventKey="ShortFilms"
          >
            Short films
          </Nav.Link>
        </Nav.Item>
        <Nav.Item>
          <Nav.Link
            onClick={() => {
              props.setMediaType("tvShow");
              props.setAttribute("other");
            }}
            eventKey="TVShows"
          >
            TV shows
          </Nav.Link>
        </Nav.Item>
        <Nav.Item>
          <Nav.Link
            onClick={() => {
              props.setMediaType("software");
              props.setAttribute("other");
            }}
            eventKey="Software"
          >
            Software
          </Nav.Link>
        </Nav.Item>
        <Nav.Item>
          <Nav.Link
            onClick={() => {
              props.setMediaType("ebook");
              props.setAttribute("other");
            }}
            eventKey="Ebooks"
          >
            E books
          </Nav.Link>
        </Nav.Item>
        <Nav.Item>
          <Nav.Link
            onClick={() => {
              props.setMediaType("favourites");
              props.setAttribute("other");
            }}
            eventKey="Favourites"
          >
            Favourites
          </Nav.Link>
        </Nav.Item>
      </Nav>
    </Container>
  );
}

export default Navigation;
