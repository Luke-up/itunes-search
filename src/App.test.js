import { render, screen } from "@testing-library/react";
import Rectangle from "./components/Rectangle";
import Square from "./components/Square.js";
import List from "./components/List.js";
import Navigation from "./components/Navigation";
import SelectorMenu from "./components/SelectorMenu.js";

//test the page display
describe("renders display layouts", () => {
  test("renders the layout with rectangle cards", () => {
    const movie = render(
      <Rectangle
        mediaType="movie"
        attribute="movieTerm"
        searchTerm="Lucy"
        limit={1}
      />
    );
    expect(movie).toMatchSnapshot();
  });
  test("renders the layout with square cards", () => {
    const software = render(
      <Square mediaType="software" searchTerm="Spotify" limit={1} />
    );
    expect(software).toMatchSnapshot();
  });
  test("renders the layout with a table", () => {
    const book = render(
      <List mediaType="ebook" searchTerm="The secret garden" limit={1} />
    );
    expect(book).toMatchSnapshot();
  });
});

//tests the navigation bar render
test("renders the navigation bar", () => {
  const screen = render(<Navigation />);
  expect(screen).toMatchSnapshot();
});

//tests functional component with different input
describe("Renders the dropdown bar correctly", () => {
  test("renders the search criteria dropdown bar", async () => {
    render(<SelectorMenu mediaType="software" />);
    expect(screen.getByRole("option")).toHaveTextContent("Search by name");
  });
  test("renders the search criteria dropdown bar", async () => {
    render(<SelectorMenu mediaType="movie" />);
    expect(screen.getByRole("combobox")).toHaveDisplayValue(
      "Choose a search criteria"
    );
  });
});
