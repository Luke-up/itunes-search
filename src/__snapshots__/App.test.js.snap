// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders display layouts renders the layout with a table 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="bg-info rounded container"
      >
        <h1
          class="text-start text-light"
        >
          Hello 
          ebook
        </h1>
        <p>
          We're looking, but haven't found anything yet :/
        </p>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="bg-info rounded container"
    >
      <h1
        class="text-start text-light"
      >
        Hello 
        ebook
      </h1>
      <p>
        We're looking, but haven't found anything yet :/
      </p>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`renders display layouts renders the layout with rectangle cards 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="bg-secondary rounded container"
      >
        <h1
          class="text-start text-light"
        >
          Hello 
          movie
        </h1>
        <p>
          We're looking, but haven't found anything yet :/
        </p>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="bg-secondary rounded container"
    >
      <h1
        class="text-start text-light"
      >
        Hello 
        movie
      </h1>
      <p>
        We're looking, but haven't found anything yet :/
      </p>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`renders display layouts renders the layout with square cards 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="bg-warning rounded container"
      >
        <h1
          class="text-start text-light"
        >
          Hello 
          software
        </h1>
        <p>
          We're looking, but haven't found anything yet :/
        </p>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="bg-warning rounded container"
    >
      <h1
        class="text-start text-light"
      >
        Hello 
        software
      </h1>
      <p>
        We're looking, but haven't found anything yet :/
      </p>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`renders successful fetch call and display layouts renders the layout with a table 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="bg-info rounded container"
      >
        <h1
          class="text-start text-light"
        >
          Hello 
          ebook
        </h1>
        <p>
          We're looking, but haven't found anything yet :/
        </p>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="bg-info rounded container"
    >
      <h1
        class="text-start text-light"
      >
        Hello 
        ebook
      </h1>
      <p>
        We're looking, but haven't found anything yet :/
      </p>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`renders successful fetch call and display layouts renders the layout with rectangle cards 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="bg-secondary rounded container"
      >
        <h1
          class="text-start text-light"
        >
          Hello 
          movie
        </h1>
        <p>
          We're looking, but haven't found anything yet :/
        </p>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="bg-secondary rounded container"
    >
      <h1
        class="text-start text-light"
      >
        Hello 
        movie
      </h1>
      <p>
        We're looking, but haven't found anything yet :/
      </p>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`renders successful fetch call and display layouts renders the layout with square cards 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="bg-warning rounded container"
      >
        <h1
          class="text-start text-light"
        >
          Hello 
          software
        </h1>
        <p>
          We're looking, but haven't found anything yet :/
        </p>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="bg-warning rounded container"
    >
      <h1
        class="text-start text-light"
      >
        Hello 
        software
      </h1>
      <p>
        We're looking, but haven't found anything yet :/
      </p>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`renders the navigation bar 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="pt-4 border rounded mx-auto my-3 container"
      >
        <div
          class="mb-4 row"
        >
          <div
            class="col"
          >
            <input
              class="form-control"
            />
          </div>
          <div
            class="col"
          >
            <select
              class="form-select"
            >
              <option>
                Search by name
              </option>
            </select>
          </div>
        </div>
        <div
          class="nav nav-tabs nav-fill"
        >
          <div
            class="nav-item"
          >
            <a
              class="nav-link"
              data-rr-ui-event-key="all"
              href="#"
              role="button"
              tabindex="0"
            >
              All
            </a>
          </div>
          <div
            class="nav-item"
          >
            <a
              class="nav-link"
              data-rr-ui-event-key="Movies"
              href="#"
              role="button"
              tabindex="0"
            >
              Movies
            </a>
          </div>
          <div
            class="nav-item"
          >
            <a
              class="nav-link"
              data-rr-ui-event-key="Podcasts"
              href="#"
              role="button"
              tabindex="0"
            >
              Podcasts
            </a>
          </div>
          <div
            class="nav-item"
          >
            <a
              class="nav-link"
              data-rr-ui-event-key="Music"
              href="#"
              role="button"
              tabindex="0"
            >
              Music
            </a>
          </div>
          <div
            class="nav-item"
          >
            <a
              class="nav-link"
              data-rr-ui-event-key="Audiobooks"
              href="#"
              role="button"
              tabindex="0"
            >
              Audiobooks
            </a>
          </div>
          <div
            class="nav-item"
          >
            <a
              class="nav-link"
              data-rr-ui-event-key="ShortFilms"
              href="#"
              role="button"
              tabindex="0"
            >
              Short films
            </a>
          </div>
          <div
            class="nav-item"
          >
            <a
              class="nav-link"
              data-rr-ui-event-key="TVShows"
              href="#"
              role="button"
              tabindex="0"
            >
              TV shows
            </a>
          </div>
          <div
            class="nav-item"
          >
            <a
              class="nav-link"
              data-rr-ui-event-key="Software"
              href="#"
              role="button"
              tabindex="0"
            >
              Software
            </a>
          </div>
          <div
            class="nav-item"
          >
            <a
              class="nav-link"
              data-rr-ui-event-key="Ebooks"
              href="#"
              role="button"
              tabindex="0"
            >
              E books
            </a>
          </div>
          <div
            class="nav-item"
          >
            <a
              class="nav-link"
              data-rr-ui-event-key="Favourites"
              href="#"
              role="button"
              tabindex="0"
            >
              Favourites
            </a>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="pt-4 border rounded mx-auto my-3 container"
    >
      <div
        class="mb-4 row"
      >
        <div
          class="col"
        >
          <input
            class="form-control"
          />
        </div>
        <div
          class="col"
        >
          <select
            class="form-select"
          >
            <option>
              Search by name
            </option>
          </select>
        </div>
      </div>
      <div
        class="nav nav-tabs nav-fill"
      >
        <div
          class="nav-item"
        >
          <a
            class="nav-link"
            data-rr-ui-event-key="all"
            href="#"
            role="button"
            tabindex="0"
          >
            All
          </a>
        </div>
        <div
          class="nav-item"
        >
          <a
            class="nav-link"
            data-rr-ui-event-key="Movies"
            href="#"
            role="button"
            tabindex="0"
          >
            Movies
          </a>
        </div>
        <div
          class="nav-item"
        >
          <a
            class="nav-link"
            data-rr-ui-event-key="Podcasts"
            href="#"
            role="button"
            tabindex="0"
          >
            Podcasts
          </a>
        </div>
        <div
          class="nav-item"
        >
          <a
            class="nav-link"
            data-rr-ui-event-key="Music"
            href="#"
            role="button"
            tabindex="0"
          >
            Music
          </a>
        </div>
        <div
          class="nav-item"
        >
          <a
            class="nav-link"
            data-rr-ui-event-key="Audiobooks"
            href="#"
            role="button"
            tabindex="0"
          >
            Audiobooks
          </a>
        </div>
        <div
          class="nav-item"
        >
          <a
            class="nav-link"
            data-rr-ui-event-key="ShortFilms"
            href="#"
            role="button"
            tabindex="0"
          >
            Short films
          </a>
        </div>
        <div
          class="nav-item"
        >
          <a
            class="nav-link"
            data-rr-ui-event-key="TVShows"
            href="#"
            role="button"
            tabindex="0"
          >
            TV shows
          </a>
        </div>
        <div
          class="nav-item"
        >
          <a
            class="nav-link"
            data-rr-ui-event-key="Software"
            href="#"
            role="button"
            tabindex="0"
          >
            Software
          </a>
        </div>
        <div
          class="nav-item"
        >
          <a
            class="nav-link"
            data-rr-ui-event-key="Ebooks"
            href="#"
            role="button"
            tabindex="0"
          >
            E books
          </a>
        </div>
        <div
          class="nav-item"
        >
          <a
            class="nav-link"
            data-rr-ui-event-key="Favourites"
            href="#"
            role="button"
            tabindex="0"
          >
            Favourites
          </a>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
